ActivatePubKey/__tests__/handler.test.ts

```diff
@@ -0,0 +1,525 @@
+    expect(findLastVersionByModelIdMock).toHaveBeenCalledWith([
+      aValidSha256AssertionRef
+    ]);
+    expect(upsertBlobFromTextMock).toHaveBeenCalledTimes(1);
+    expect(upsertBlobFromTextMock).toHaveBeenCalledWith(
+      blobServiceMock,
+      LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME,
+      `${aValidActivatePubKeyPayload.fiscal_code}-${aValidSha256AssertionRef}`,
+      ""
+    );
+    expect(upsertMock).toHaveBeenCalledTimes(2);
+    expect(upsertMock).toHaveBeenNthCalledWith(1, {
+      pubKey: aPendingRetrievedPopDocument.pubKey,
+      // the assertion Ref for masterKey is created by the getAllAssertionRefs method
+      assertionRef: assertionRefsResult.right.master,
+      assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}`,
+      status: PubKeyStatusEnum.VALID,
+      assertionType: aValidActivatePubKeyPayload.assertion_type,
+      fiscalCode: aValidActivatePubKeyPayload.fiscal_code,
+      expiredAt: aValidActivatePubKeyPayload.expires_at
+    });
+    expect(upsertMock).toHaveBeenNthCalledWith(2, {
+      pubKey: aPendingRetrievedPopDocument.pubKey,
+      assertionRef: aValidSha256AssertionRef,
+      assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}`,
+      status: PubKeyStatusEnum.VALID,
+      assertionType: aValidActivatePubKeyPayload.assertion_type,
+      fiscalCode: aValidActivatePubKeyPayload.fiscal_code,
+      expiredAt: aValidActivatePubKeyPayload.expires_at
+    });
+
+    expect(res.kind).toBe("IResponseSuccessJson");
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessJson",
+      value: retrievedLollipopKeysToApiActivatedPubKey(
+        aValidRetrievedPopDocument
+      )
+    });
+  });
+
+  it("should success given valid informations when used algo == master algo", async () => {
+    upsertBlobFromTextMock.mockImplementationOnce(() =>
+      Promise.resolve(
+        E.right(O.fromNullable({ name: "blob" } as BlobService.BlobResult))
+      )
+    );
+
+    findLastVersionByModelIdMock.mockImplementationOnce(() =>
+      TE.right(O.some(aPendingRetrievedPopDocumentWithMasterAlgo))
+    );
+
+    upsertMock.mockImplementationOnce(() => {
+      return TE.right(aValidRetrievedPopDocumentWithMasterAlgo);
+    });
+
+    const activatePubKeyHandler = ActivatePubKeyHandler(
+      getPopDocumentReader(lollipopPubKeysModelMock),
+      getPopDocumentWriter(lollipopPubKeysModelMock),
+      getAssertionWriter(
+        blobServiceMock,
+        LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME
+      )
+    );
+
+    const aValidActivatePubKeyPayload: ActivatePubKeyPayload = {
+      fiscal_code: aFiscalCode,
+      expires_at: new Date(),
+      assertion_type: AssertionTypeEnum.SAML,
+      assertion: "" as NonEmptyString
+    };
+
+    const assertionRefsResult = await getAllAssertionsRef(
+      JwkPubKeyHashAlgorithmEnum.sha512,
+      JwkPubKeyHashAlgorithmEnum.sha512,
+      aValidJwk
+    )();
+
+    if (assertionRefsResult._tag === "Left") fail();
```

Comment:
why not asserting a Right here?

Reply:
using fail, we enable the use of `assertionRefsResult` `right` property below without an if. removing this will need a snippet below similar to 
```ts
if(E.isRight(var)){
...
var.right
...
}
```

Reply:
Ok, expect is not going to do any type narrowing, and with this code you're removing a nested block. LGTM.